ntbs-integration-tests/Helpers/WebApplicationExtensions.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using ntbs_integration_tests.TestServices;
+using ntbs_service;
+using ntbs_service.Models;
+using ntbs_service.Services;
+
+namespace ntbs_integration_tests.Helpers
+{
+    public static class WebApplicationExtensions
+    {
+        public static WebApplicationFactory<Startup> WithNhsUserBuilder(this WebApplicationFactory<Startup> factory, IDictionary<int, string> idServiceCodeMapping)
+        {
+            return factory.WithWebHostBuilder(builder =>
+            {
+                SetupDatabase(builder, idServiceCodeMapping);
+                builder.ConfigureTestServices(services =>
+                {
+                    services.AddScoped<IUserService, NhsUserService>();
+                });
+            });
+        }
+
+        public static WebApplicationFactory<Startup> WithPheUserBuilder(this WebApplicationFactory<Startup> factory,
+                                                                        IDictionary<int, string> idServiceCodeMapping,
+                                                                        IDictionary<int, string> idPostcodeMapping)
+        {
+            return factory.WithWebHostBuilder(builder =>
+            {
+                SetupDatabase(builder, idServiceCodeMapping, idPostcodeMapping);
+                builder.ConfigureTestServices(services =>
+                {
+                    services.AddScoped<IUserService, PheUserService>();
+                });
+            });
+        }
+
+        public static HttpClient WithoutRedirects(this WebApplicationFactory<Startup> factory)
+        {
+            return factory.CreateClient(new WebApplicationFactoryClientOptions
+            {
+                AllowAutoRedirect = false

```

Comment:
What are these redirects that we're turning off?  Is it something quite typical for tests, or something in our code that prompts it?

Reply:
By default redirects will happen, so that you can't check as easily that a page gave you a redirect result (you'd just get a status ok on the page you were redirected to). I think it's clearer to check for the explicit redirect

Reply:
oh, that makes a lot of sense - isolates it more too, and i imagine the tests are faster.
Thanks for explaining! :)